grails-app/controllers/au/org/emii/portal/WpsController.groovy

```diff
@@ -1,31 +1,50 @@
 package au.org.emii.portal
-
-import au.org.emii.portal.proxying.ExternalRequest
 import au.org.emii.portal.proxying.RequestProxyingController
-
 import org.joda.time.DateTime
 
 class WpsController extends RequestProxyingController {
 
     def wpsService
 
+    // Users get WPS status here -> HTML
     def jobReport = {
-        params.url = _getExecutionStatusUrl(params)
-        _performProxyingIfAllowed()
+        params.url = wpsService._getExecutionStatusUrl(params)
+        def execResponse = _performProxyingIfAllowed()
+        if (execResponse != null) {
+            if (execResponse.name() == "ExceptionReport") {
+                _renderExecutionFailed(execResponse)
+            }
+            else if (execResponse.ProcessOutputs.Output.Reference.@href != null){
+                params.status = "Preparing download"
+                _renderExecutionStatus(execResponse)
+            }
+            else {
+                params.status = "Download ready"
+                _renderExecutionStatus(execResponse)
+            }
+        }
     }
 
+    // the WPS geoserver plugin callback -> email
     def jobComplete = {
-        params.email.subject = "IMOS subsetting complete - ${params.uuid}"
-        params.email.template = 'jobComplete'
-        wpsService.notifyViaEmail(params)
 
+        params.url = wpsService._getExecutionStatusUrl(params)
+        def execResponse = _performProxyingIfAllowed()
+        if (execResponse != null) {
+            if (execResponse.name() == "ExceptionReport") {
+                 wpsService._notifyErrorViaEmail(params)
+            }
+            else {
+                 wpsService._notifyDownloadViaEmail(params)
+            }
+        }
         render status: 200
```

Comment:
I know you didn't write this, but this is a good opportunity to change this 200 literal value. We have constants defined for the HTTP status codes which we use throughout the Portal. See LayerController for an example.